packages/components/src/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { ReactElement, cloneElement, useState } from "react";
+import ReactDatePicker from "react-datepicker";
+import { XOR } from "ts-xor";
+import { Button } from "../Button";
+import { strFormatDate } from "../FormatDate";
+
+/**
+ * Disabling no-internal-modules here because we need
+ * to reach into the package to get the css file.
+ */
+// eslint-disable-next-line import/no-internal-modules
+import "react-datepicker/dist/react-datepicker.css";
+
+interface DatePickerReturnedDates {
+  readonly formatted: string;
+  readonly raw: Date;
```

Comment:
Is there a reason why `raw` isn't optional?  Just wondering!

Reply:
For this one, this will be the returned value from the `onChange` prop. We would use it like this.

```tsx
function Component() {
 return <Datepicker onChage={handleChange}
 
 function handleChange(val: DatePickerReturnedDates) {
   const { formatted, raw } = val;
 }
}

Reply:
Ah okay gotcha!  Was just wondering (more for my own knowledge) why it wasn't like:
```  readonly raw?: Date;
```

Reply:
It is becuase this is always returned fro, the `onChange`. You can see in the `handleChange` we do this.

```tsx
function handleChange(val: Date) {
    ...
    onChange && onChange({ raw: val, formatted: strFormatDate(val) });
}
```

Here we **always** return the `raw` and the `formatted` values. So when we set the interface as:

```tsx
interface DatePickerReturnedDates {
  readonly formatted: string;
  readonly raw: Date;
}
```

We are letting the consumer know that these will always be returned from the `onChange` function. If we were to do something like:


```tsx
interface DatePickerReturnedDates {
  readonly formatted: string;
  readonly raw?: Date;
}
```

We would be saying "You will always get the `formatted` date, but you might not always get the `raw` date" which would be untrue.

Reply:
Okay, that makes sense!  Thanks for that :)